libs/libarchfpga/src/arch_util.cpp

```diff
@@ -1068,36 +1125,151 @@ void CreateModelLibrary(t_arch* arch) {
     model_library[3].index = 3;
+    model_library[3].inputs = new t_model_ports[2];
 
-    model_library[3].inputs = new t_model_ports[1];
     model_library[3].inputs[0].dir = IN_PORT;
-    model_library[3].inputs[0].name = vtr::strdup("in");
-    model_library[3].inputs[0].next = nullptr;
+    model_library[3].inputs[0].name = vtr::strdup("D");
+    model_library[3].inputs[0].next = &model_library[3].inputs[1];
     model_library[3].inputs[0].size = 1;
     model_library[3].inputs[0].min_size = 1;
     model_library[3].inputs[0].index = 0;
     model_library[3].inputs[0].is_clock = false;
-    model_library[3].inputs[0].combinational_sink_ports = {"out"};
+    model_library[3].inputs[0].clock = "clk";
+    model_library[3].inputs[0].trigg_edge = TriggeringEdge::FALLING_EDGE;
+
+    model_library[3].inputs[1].dir = IN_PORT;
+    model_library[3].inputs[1].name = vtr::strdup("clk");
+    model_library[3].inputs[1].next = nullptr;
+    model_library[3].inputs[1].size = 1;
+    model_library[3].inputs[1].min_size = 1;
+    model_library[3].inputs[1].index = 0;
+    model_library[3].inputs[1].is_clock = true;
+    model_library[3].inputs[1].trigg_edge = TriggeringEdge::FALLING_EDGE;
 
     model_library[3].instances = nullptr;
-    model_library[3].next = nullptr;
+    model_library[3].next = &model_library[4];
 
     model_library[3].outputs = new t_model_ports[1];
     model_library[3].outputs[0].dir = OUT_PORT;
-    model_library[3].outputs[0].name = vtr::strdup("out");
+    model_library[3].outputs[0].name = vtr::strdup("Q");
     model_library[3].outputs[0].next = nullptr;
     model_library[3].outputs[0].size = 1;
     model_library[3].outputs[0].min_size = 1;
     model_library[3].outputs[0].index = 0;
     model_library[3].outputs[0].is_clock = false;
+    model_library[3].outputs[0].clock = "clk";
+    model_library[3].outputs[0].trigg_edge = TriggeringEdge::FALLING_EDGE;
+
+    //NAMES
+    model_library[4].name = vtr::strdup(MODEL_NAMES);
+    model_library[4].index = 4;
+
+    model_library[4].inputs = new t_model_ports[1];
+    model_library[4].inputs[0].dir = IN_PORT;
+    model_library[4].inputs[0].name = vtr::strdup("in");
+    model_library[4].inputs[0].next = nullptr;
+    model_library[4].inputs[0].size = 1;
+    model_library[4].inputs[0].min_size = 1;
+    model_library[4].inputs[0].index = 0;
+    model_library[4].inputs[0].is_clock = false;
+    model_library[4].inputs[0].combinational_sink_ports = {"out"};
+
+    model_library[4].instances = nullptr;
+    model_library[4].next = nullptr;
+
+    model_library[4].outputs = new t_model_ports[1];
+    model_library[4].outputs[0].dir = OUT_PORT;
+    model_library[4].outputs[0].name = vtr::strdup("out");
+    model_library[4].outputs[0].next = nullptr;
+    model_library[4].outputs[0].size = 1;
+    model_library[4].outputs[0].min_size = 1;
+    model_library[4].outputs[0].index = 0;
+    model_library[4].outputs[0].is_clock = false;
```

Comment:
Duplicating the whole latch to add a triggering polarity is pretty ugly.  Is there a way to avoid this?  If it's unavoidable or driven by some difficult to fix constraint, should comment why.

Reply:
Sadly, I don't see any possibility for avoiding this. I explained the issue in comments.